### PR TITLE
[tensorboard] Cleanup API and remove 'experimental' warning

### PR DESCRIPTION
**Pull request by orionr**
This cleans up the `torch.utils.tensorboard` API to remove all kwargs usage (which isn't clear to the  user) and removes the "experimental" warning in prep for our 1.2 release.

We also don't need the additional PyTorch version checks now that we are in the codebase itself.

cc @ezyang @lanpa @natalialunova

### CODE REVIEW

**Comment by natalialunova** [comment]
Looks good!

File: torch/utils/tensorboard/writer.py [hunk 1]
```diff
@@ -664,15 +653,12 @@ def add_graph(self, model, input_to_model=None, verbose=False, **kwargs):
             )
             if isinstance(model, list):
                 if isinstance(model[0], core.Net):
-                    current_graph = nets_to_graph_def(
-                        model, **kwargs)
```
Comment (lanpa) [inline torch/utils/tensorboard/writer.py]: There is no test for caffe's graph yet in pytorch repo, so I suggest leave it as is.

Comment (orionr) [reply torch/utils/tensorboard/writer.py]: In that case, let's add some tests before I land this. I'll file a separate task.

Comment (lanpa) [reply torch/utils/tensorboard/writer.py]: Accidentally opened a new branch: https://github.com/pytorch/pytorch/tree/remove-tensorboard-experimental

**Comment by orionr** [comment]
Tests passed even though not updated in GitHub UI.

**Comment by ailzhang** [comment]
FYI this PR might have broken asan test.

**Comment by orionr** [comment]
Unfortunately looks like it. We can revert (and re-land) or wait to see if https://github.com/pytorch/pytorch/pull/23000 fixes it.

**Comment by orionr** [comment]
Actually, looks like there are other issues, so reverting.

**Pull request closed without merging**